src/config_file.c

```diff
@@ -524,15 +689,59 @@ gboolean save_slot_status(RaucSlot *dest_slot, RaucImage *mfimage, GError **erro
 
 free:
 	g_clear_pointer(&slotstatuspath, g_free);
-	g_clear_pointer(&slot_state, free_slot_status);
 
 	return res;
 }
 
+static gboolean save_slot_status_globally(RaucSlot *dest_slot, GError **error) {
+	GKeyFile *key_file = g_key_file_new();
+	GError *ierror = NULL;
+	gboolean res;
+	gchar *group;
+
+	g_return_val_if_fail(dest_slot, FALSE);
+	g_return_val_if_fail(dest_slot->status, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+	g_return_val_if_fail(r_context()->config->statusfile_path, FALSE);
+
+	g_key_file_load_from_file(key_file, r_context()->config->statusfile_path, G_KEY_FILE_NONE, &ierror);
```

Comment:
I'd expect the global slot status file to be regenerated from the in-memory-data. That would avoid having to read it every time. We should drop the dest_slot argument here as well.

Reply:
That's okay with me and as you already added commit a39dcb10dc9f ("src/config_file: save status data for all slots when using a global file"), I consider your change request as done.